util.js

```diff
@@ -236,13 +236,21 @@
         customFavicon = domain.match(faviconDomains);
 
         if (customFavicon) {
-            // check for dark background & available variant
-            if (isDarkBg && darkVariants.test(customFavicon)) {
-                variant = '.white';
+            // validate domain
+            customFavicon = customFavicon.toString();
+            // if the string matches a favicon domain, remove the TLD (.com) and ensure we match on the domain name only.
+            tokens = sourceUrl.split('.');
+            domainMatch = (tokens.length > 1) ? (tokens[tokens.length - 2] === customFavicon) : true;
+          
+            if (domainMatch) {
+                // check for dark background & available variant
+                if (DDG.settings && DDG.settings && DDG.settings.updater && DDG.settings.updater.isDarkBg && darkVariants.test(customFavicon)) {
```

Comment:
You've got `DDG.settings` twice there now. I'm just gonna quickly remove one of them before merging.